software/firmware/europi.py

```diff
@@ -443,10 +451,31 @@ def centre_text(self, text):
 
         padding_top = (self.height - (len(lines) * 9)) / 2
         for index, content in enumerate(lines):
-            x_offset = int((self.width - ((len(content) + 1) * 7)) / 2) - 1
             y_offset = int((index * 9) + padding_top) - 1
-            self.text(content, x_offset, y_offset)
-        oled.show()
+            self.centred_text_line(content, y_offset)
+        self.show()
+
+    def centred_text_line(self, text, y, colour=1):

```

Comment:
[discussion] The utility of exposing this method in the firmware is not clear to me. Again, likely because there are no external calls to it. 

Also, I'm hesitant to add additional text methods to the API without a clear goal. It may be worth investigating all of the ways that scripts are putting text on the screen to identify the actual needs of our users and where the firmware is falling short. That way we can design a text api that is cohesive and meets the needs of our users.

Reply:
Same idea as for the `arrow()` method, I'm already using it in my script so I thought I could add it to this class for everyone to use it.
Centering a text on a single line sounds like a pretty generic thing.

For both of these methods, I can move them back in my script if necessary, it just seems to me that this class could be more useful than what it is now if it would contain more useful methods.

Reply:
> Centering a text on a single line sounds like a pretty generic thing.

For sure, I just think that a larger text API deserves some thought. And we might paint ourselves into a corner if we add a few text functions to the firmware without doing that work first.